src/torrent.cpp

```diff
@@ -4728,6 +4725,9 @@ namespace {
 		// ever downloaded for it.
 		m_picker->restore_piece(j->piece);
 
+		if (m_ses.alerts().should_post<hash_failed_alert>())
```

Comment:
did you make sure ``on_piece_sync`` isn't used as a callback in any other situation? like where there was no piece failure?

Reply:
Yes.